### PR TITLE
(FMT): Code samples

### PR DESCRIPTION
**Pull request by jajakobyly**
This PR adds better code samples to the code style settings page. Also I have tinkered around type params formatting.

### CODE REVIEW

File: src/main/resources/org/rust/ide/formatter/settings/blank_lines_sample.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+        }
+
+
+
+
+        // Close the opened bracket and return a fmt::Result value
+        write!(f, "]")
+    }
+}
```
Comment (matklad) [inline src/main/resources/org/rust/ide/formatter/settings/blank_lines_sample.rs]: Why are these sample are misformated so drastically? 

I think we should leave one instance with too many lines, and one with too few. That way, the user can easily identify what exactly is changed when the option is toggled.

File: src/main/kotlin/org/rust/ide/formatter/impl/utils.kt [hunk 2]
```diff
@@ -48,24 +47,22 @@ val MACRO_ARGS = ts(MACRO_ARG, FORMAT_MACRO_ARGS, TRY_MACRO_ARGS)
 
 val FN_DECLS = ts(FN_ITEM, FOREIGN_FN_DECL, TRAIT_METHOD_MEMBER, IMPL_METHOD_MEMBER, BARE_FN_TYPE, LAMBDA_EXPR)
 
-val FN_SHARED_ALIGN_OWNERS = orSet(PARAMS_LIKE, ts(RET_TYPE, WHERE_CLAUSE))
-
 
 val PsiElement.isTopLevelItem: Boolean
-    get() = (this is RustItemElement || this is RustAttrElement) && this.parent is RustFile
+    get() = (this is RustItemElement || this is RustAttrElement) && parent is RustMod
```
Comment (matklad) [inline src/main/kotlin/org/rust/ide/formatter/impl/utils.kt]: Not related to this PR, but the second case `this is RustAttrElement` seems suspicious. The parent of `RustAttr` is just another item. Perhaps this should be `RustInnerAttr`, to handle mod inner attributes?

File: src/test/kotlin/org/rust/ide/formatter/RustFormatterTestCase.kt [hunk 3]
```diff
@@ -52,6 +52,11 @@ class RustFormatterTestCase : FormatterTestCase() {
         doTest()
     }
 
+    fun testAlignTypeParamsOn() {
+        custom().ALIGN_TYPE_PARAMS = true
+        doTest()
+    }
```
Comment (matklad) [inline src/test/kotlin/org/rust/ide/formatter/RustFormatterTestCase.kt]: Again, not related to PR, but a more DSLy API would be

```Kotlin
   fun testAlignTypeParamsOn() = custom {
       ALIGN_TYPE_PARAMS = true
   }
```

where `custom` is defined as

```Kotlin
fun custom(block: RustCustomSettings.() -> Unit) {
    val settings = settings.getCustomSettings(RustCodeStyleSettings::class.java)
    settings.block() // <- this is pure awesomeness, because the extension function here is a parameter. 
    doTest()
}
```

File: src/main/kotlin/org/rust/ide/formatter/impl/utils.kt (hunk 2 above)
Comment (jajakobyly) [inline src/main/kotlin/org/rust/ide/formatter/impl/utils.kt]: Yeah, I haven't noticed this. It was correct when parent could be only a file. Fix it here?

Comment (matklad) [inline src/main/kotlin/org/rust/ide/formatter/impl/utils.kt]: I'd rather make a separate PR, as usual, but if you find it more convenient to fix it here go ahead.

Comment (jajakobyly) [inline src/main/kotlin/org/rust/ide/formatter/impl/utils.kt]: I can submit separate PR with further small improvements again in few days, it's no problem for me.
